torch_brain/nn/lora.py

```
@@ -0,0 +1,602 @@
+        # Get original layer parameters
+        original_weight = linear_layer.weight.data
+        original_bias = linear_layer.bias
+
+        # Calculate dimensions
+        in_features = linear_layer.in_features
+        out_features = linear_layer.out_features
+
+        if out_features % num_projections != 0:
+            raise ValueError(
+                f"Output features ({out_features}) must be divisible by number of projections ({self.num_projections})"
+            )
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)